File: neoforge/src/main/java/mconi/mixins/neoforge/client/SplashManagerMixin.java
/*
 *    This file is part of the minecraft-ONI-mod
 *    licensed under the GNU GPL v3 License.

 *    Copyright (C) 2024  Leander Knüttel
 *
 *    This program is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as published by
 *    the Free Software Foundation, either version 3 of the License, or
 *    (at your option) any later version.
 *
 *    This program is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package mconi.mixins.neoforge.client;

import net.minecraft.client.gui.components.SplashRenderer;
import net.minecraft.network.chat.Component;
import net.minecraft.client.resources.SplashManager;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;

/**
 * @author Leander Knüttel
 * @version 17.05.2024
 */
@Mixin(SplashManager.class)
public class SplashManagerMixin {
    @Inject(method = "getSplash", at = @At("HEAD"), cancellable = true)
    private void setSplash(CallbackInfoReturnable<SplashRenderer> cir) {
        cir.setReturnValue(new SplashRenderer(Component.literal("§c§lExampleMod NeoForge!")));
    }
}

File: fabric/src/main/kotlin/conservecraft/mixins/fabric/client/SplashManagerMixin.kt
package conservecraft.mixins.fabric.client

import net.minecraft.client.gui.components.SplashRenderer
import net.minecraft.client.resources.SplashManager
import net.minecraft.network.chat.Component
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable

@Mixin(SplashManager::class)
class SplashManagerMixin {
    @Inject(method = ["getSplash"], at = [At("HEAD")], cancellable = true)
    private fun setSplash(cir: CallbackInfoReturnable<SplashRenderer>) {
        cir.returnValue = SplashRenderer(Component.literal("§c§lExampleMod Fabric!"))
    }
}
